patches/tModLoader/Terraria/NPC.cs.patch

```diff
@@ -51,8 +51,84 @@
++	/// Used to keep vanilla and modded gravity effects working neatly
++	/// </summary>
++	private float internalGravityBase = 0.3f;
++
++	/// <summary>
++	/// Modify this value in order to change the NPCs active gravity, this can be done in AI as gravity values are reset slightly beforehand, and used slightly after.
++	/// </summary>
++	public MultipliableFloat Gravity = MultipliableFloat.One;//TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// The current actual change in velocity due to gravity applied every frame. Returns the combination of tMod internal values and <see cref="Gravity"/> that will actually be used to modify gravity. Do not use unless there is a specific reason why as this may causes issues with other mods, use <see cref="Gravity"/> instead.
++	/// </summary>
+-	private static float gravity = 0.3f;
++	//private static float gravity = 0.3f;
++	public float gravity {
++		get => internalGravityBase * Gravity.Value;
++		private set {
++			Gravity = MultipliableFloat.One;
++			internalGravityBase = value;
++		}
++	}
++
++	/// <summary>
++	/// Used to keep vanilla and modded gravity effects working neatly
++	/// </summary>
++	private float internalMaxFallSpeedBase = 10f; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// Modify this value in order to change the NPCs active maxFallSpeed, this can be done in AI as gravity values are reset slightly beforehand
++	/// </summary>
++	public MultipliableFloat MaxFallSpeed = MultipliableFloat.One; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// The current actual fall speed cap in velocity applied every frame. Returns the combination of tMod internal values and <see cref="MaxFallSpeed"/> that will actually be used to modify velocity. Do not use unless there is a specific reason why as this may causes issues with other mods, use <see cref="MaxFallSpeed"/> instead.
++	/// </summary>
++	public float maxFallSpeed { //TML: #AccessibleNPCGravity, added
++		get => internalMaxFallSpeedBase * MaxFallSpeed.Value;
++		set {
++			MaxFallSpeed = MultipliableFloat.One;
++			internalGravityBase = value;
++		}
++	}
++
++	/// <summary>
++	/// The effect of the types of wet on the NPCs gravity, here for reference when modifying gravity:
++	/// <br> wet, honetWet, and shimmerWet.</br>
++	/// <br> These are mutually exclusive and in reverse order priority where lava is the same as water. </br>
++	///	</summary>
++	public static float[] GravityWetMultipliers => new float[] { 2f / 3f, 1f / 3f, 1.5f / 3f }; //TML: #AccessibleNPCGravity, added
++
++	/// <summary>
++	/// The effect of the types of wet on the NPCs maxFallSpeed, here for reference when modifying maxFallSpeed:
++	/// <br> wet, honetWet, and shimmerWet.</br>
++	/// <br> These are mutually exclusive and in reverse order priority where lava is the same as water. </br>
++	///	</summary>
++	public static float[] MaxFallSpeedWetMultipliers => new float[] { 0.7f, 0.4f, 0.55f }; //TML: #AccessibleNPCGravity, added
```

Comment:
As noted earlier, I think these should line up with `LiquidID`.

Also, I know you don't want people to modify them, but eh, allocating a new array on every access is not the way to go. Better to just advise mods not to modify. You could consider making the field type `RefReadOnlyArray<float>` which will prevent easy modification

Reply:
How about this?
```
private static float[] _maxFallSpeedWetMultipliers = new float[4] { 0.7f, 0.7f, 0.4f, 0.55f };
public static float[] MaxFallSpeedWetMultipliers => _maxFallSpeedWetMultipliers;
```

Reply:
Hm, lets do 
```cs
public static RefReadOnlyArray<float> MaxFallSpeedWetMultipliers { get; } = new float[] { ... };
```

Reply:
Yeah, dunno why I wrote in the backing fields manually, this is why I shouldn't code when dead tired 🤦‍♂️

Reply:
Fixed in commit